src/back/hlsl/writer.rs

```diff
@@ -0,0 +1,20 @@
+use crate::back::hlsl::Error;
```

Comment:
I don't think any backend should include anything from either `crate::back` or `crate::front`. Instead, let's do `use super::Error`.

Reply:
Yes agreed, probably an auto complete function in my IDE that auto imports.